portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java

```diff
@@ -1,33 +1,104 @@
+    private String dateAndTime() {
+        SimpleDateFormat formatter= new SimpleDateFormat("yyyy-MM-dd 'at' HH:mm:ss 'ET'");
+        Date date = new Date(System.currentTimeMillis());
+
+        return formatter.format(date);
+    }
+
+    // Send data to datastore
+    private void store(String name, String email, String number, String message, String date) {
+        Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+        KeyFactory keyFactory = datastore.newKeyFactory().setKind("Contacts");
+        FullEntity contactEntity =
+            Entity.newBuilder(keyFactory.newKey())
+                .set("name", name)
+                .set("email", email)
+                .set("number", number)
+                .set("message", message)
+                .set("date", date)
+                .build();
+        datastore.put(contactEntity);
+    }
+
+    // checks if data submitted is not null and larger than 5 characters (2 for name)
+    private Boolean checkIfValid(String message, String number, String email, String name) {
+        if (message == null || message.length() < 5) {
+            System.out.println("Message is too short");
+            return false;
+        } 
+        if (number == null || number.length() < 5) {
+            System.out.println("Number is too small");
+            return false;
+        }
+        if (email == null || email.length() < 5) {
+            System.out.println("Email is too small");
+            return false;
+        }
+        if (name == null || name.length() < 2) {
+            System.out.println("Name is too small");
+            return false;
+        }
+
+        return true;
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        // Get the value entered in the form.
+        String message = request.getParameter("message");
+        String number = request.getParameter("number");
+        String email = request.getParameter("email");
+        String name = request.getParameter("name");
 
-    // Get the value entered in the form.
-    String message = request.getParameter("message");
-    String number = request.getParameter("number");
-    String email = request.getParameter("email");
-    String name = request.getParameter("name");
+        Boolean dataIsSafe = checkIfValid(message, number, email, name);
 
 
-    // Print the value so you can see it in the server logs.
-    System.out.println("------------------------");
-    System.out.println("Name: " + name);
-    System.out.println("Email: " + email);
-    System.out.println("Number " + number);
-    System.out.println("\nMessage: " + message);
-    System.out.println("------------------------");
+        if (dataIsSafe) {
+            // readable system time
+            String readableDate = dateAndTime();
+    
+            // Prints values to console
+            logMessage(name, email, number, message, readableDate);
```

Comment:
From a debugging standpoint, it might be worth always logging the message whether it's safe or not.